cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsInstanceConnector.java

```diff
@@ -163,33 +164,37 @@ public String getConsoleOutput(AuthenticatedContext authenticatedContext, CloudI
                 .map(status -> status.getCloudInstance()).collect(Collectors.toList());
     }
 
-    private void doReboot(List<CloudInstance> affectedVMs, AmazonEC2Client amazonEC2Client, List<CloudInstance> instances) {
+    private List<CloudVmInstanceStatus> doStart(AuthenticatedContext ac, List<CloudInstance> instances) {
+        List<CloudVmInstanceStatus> rebootedVmsStatus = new ArrayList<>();
         for (CloudInstance instance: instances) {
             try {
-                amazonEC2Client.rebootInstances(new RebootInstancesRequest().withInstanceIds(List.of(instance.getInstanceId())));
-                affectedVMs.add(instance);
+                rebootedVmsStatus.addAll(start(ac, null, List.of(instance)));
             } catch (AmazonEC2Exception e) {
-                LOGGER.warn(String.format("Unable to reboot instance %s", instance), e);
+                LOGGER.warn(String.format("Unable to start instance %s", instance), e);
             }
         }
+        return rebootedVmsStatus;
     }
 
-    private void doStart(List<CloudInstance> affectedVMs, AuthenticatedContext ac, List<CloudInstance> instances) {
+    private void doStop(AuthenticatedContext ac, List<CloudInstance> instances) {
         for (CloudInstance instance: instances) {
             try {
-                start(ac, null, List.of(instance));
-                affectedVMs.add(instance);
+                stop(ac, null, List.of(instance));
             } catch (AmazonEC2Exception e) {
-                LOGGER.warn(String.format("Unable to start instance %s", instance), e);
+                LOGGER.warn(String.format("Unable to stop instance %s", instance), e);
             }
         }
     }
 
-    private void logInvalidStatuses(List<CloudVmInstanceStatus> instances) {
-        for (CloudVmInstanceStatus instance: instances) {
-            LOGGER.warn(String.format("Unable to reboot instance %s because of invalid status %s.",
-                    instance.getCloudInstance().getInstanceId(), instance.getStatus().toString()));
+    public void logInvalidStatuses(List<CloudVmInstanceStatus> instances, InstanceStatus targetStatus) {
+        if (instances == null || instances.size() < 1) {
+            return;
```

Comment:
multiple issues here: 
- please don't use return like this, if possible avoid in the middle of returns in methods, or use `else` to indicate there are multiple exit points in the method
- if you would like to check if a collection is empty, please use `isEmpty()`
- I would negate the if condition and put the body of the method inside. You can use `CollectionUtils` from apache (`CollectionUtils.isNotEmpty(instances)`) or spring (`!CollectionUtils.isEmpty(instances)`) or write the whole condition by hand (`instances != null && !instances.isEmpty()`). It would still result in the same logic but there won't be that unnecessary return